src/worker/predictive-text-studio-worker-impl.ts

```diff
@@ -77,8 +81,9 @@ export class PredictiveTextStudioWorkerImpl
         new Error("Cannot find any files in the IndexedDB")
       );
     } else {
-      const kmpFile = await linkStorageToKmp(this.storage);
-      this._emitPackageCompileSuccess(kmpFile);
+      const KMPArrayBuffer = await linkStorageToKmp(this.storage);
+      this.saveKMPPackage(KMPArrayBuffer);
+      this._emitPackageCompileSuccess();
```

Comment:
It doen not matter in this project, but when you are creating a public API, this would be considered a breaking change and there's really nothing stopping you from preserving backwards compatibility. You can definitely both save the `KMP` packlage and send it to the `onPackageCompileSuccess()` callback.

Again, in this project it doesn't matter, but something to consider when making APIs. If you wanted people to use `getKMPPackage()` instead, you could add to the documentation that this parameter is deprecated.

```typescript
      this._emitPackageCompileSuccess(kmp);
```

Reply:
Thank you for your suggestion!